molgenis-apps/src/main/resources/js/apps.js

```diff
@@ -10,7 +10,7 @@
                 onSubmitSuccess: function () {
                     window.location = molgenis.contextUrl;
                 }
-            }), $('#form-modal-container')[0]).setState({showModal: true});
+            }), $('<div>')[0]);
```

Comment:
The React component is rendered in a detached DOM element which causes the component to be recreated on every button select. The same strategy is used in the user manager and scripts plugin.